service/policy/attributes/attributes_test.go

```diff
@@ -0,0 +1,284 @@
+package attributes
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/bufbuild/protovalidate-go"
+	"github.com/opentdf/platform/protocol/go/policy"
+	"github.com/opentdf/platform/protocol/go/policy/attributes"
+	"github.com/stretchr/testify/require"
+)
+
+func getValidator() *protovalidate.Validator {
+	v, err := protovalidate.New()
+	if err != nil {
+		panic(err)
+	}
+	return v
+}
+
+var (
+	validName   = "name"
+	validValue1 = "value1"
+	validValue2 = "value_2"
+	validValue3 = "3_value"
+	validUUID   = "00000000-0000-0000-0000-000000000000"
+)
+
+// Create Attributes (definitions)
+
+func TestCreateAttribute_Valid_Succeeds(t *testing.T) {
+	req := &attributes.CreateAttributeRequest{
+		Name:        validName,
+		NamespaceId: validUUID,
+		Rule:        policy.AttributeRuleTypeEnum_ATTRIBUTE_RULE_TYPE_ENUM_ALL_OF,
+	}
+
+	v := getValidator()
+	err := v.Validate(req)
+
+	require.NoError(t, err)
+}
+
+func TestCreateAttribute_WithValues_Valid_Succeeds(t *testing.T) {
+	req := &attributes.CreateAttributeRequest{
+		Name:        validName,
+		NamespaceId: validUUID,
+		Rule:        policy.AttributeRuleTypeEnum_ATTRIBUTE_RULE_TYPE_ENUM_ALL_OF,
+		Values: []string{
+			validValue1,
+			validValue2,
+			validValue3,
+		},
+	}
+
+	v := getValidator()
+	err := v.Validate(req)
+
+	require.NoError(t, err)
+}
+
+func TestCreateAttribute_NameTooLong_Fails(t *testing.T) {
+	name := strings.Repeat("a", 254)
+	req := &attributes.CreateAttributeRequest{
+		Name:        name,
+		NamespaceId: validUUID,
+		Rule:        policy.AttributeRuleTypeEnum_ATTRIBUTE_RULE_TYPE_ENUM_ALL_OF,
+	}
+
+	v := getValidator()
+	err := v.Validate(req)
+
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "string.max_len")
+}
+
+func TestCreateAttribute_NameWithSpace_Fails(t *testing.T) {
+	req := &attributes.CreateAttributeRequest{
+		Name:        "invalid name",
+		NamespaceId: validUUID,
+		Rule:        policy.AttributeRuleTypeEnum_ATTRIBUTE_RULE_TYPE_ENUM_ALL_OF,
+	}
+
+	v := getValidator()
+	err := v.Validate(req)
+
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "[attribute_name_format]")
+}
+
+func TestCreateAttribute_NameWithNonAlphanumeric_Fails(t *testing.T) {
+	names := []string{
+		"invalid@name",
+		"invalid#name",
+		"invalid$name",
+		"invalid%name",
+		"invalid^name",
+		"invalid:name",
+		"invalid/name",
+		"invalid&name",
```

Comment:
I think this is fine, but I'm not sure we need to test everything since everything outside the set is significantly larger than in the set.

Reply:
True, and that's worth clarifying in a comment. Updated commit: [04eef36](https://github.com/opentdf/platform/pull/536/commits/04eef36ef30f84e3abad08a54693ae8f06f8a39d)